PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java

```diff
@@ -185,6 +194,10 @@ private void createUIComponents() {
         yarnEndpointField = new HintTextField("(Optional)Example: http://hn0-spark2:8088");
     }
 
+    public SparkClusterType getSparkClusterType() {
+        return isHDInsightClusterSelected() ? SparkClusterType.HDINSIGHT_CLUSTER : SparkClusterType.LIVY_LINK_CLUSTER;
```

Comment:
Only two types?

Reply:
We override `getSparkClusterType` method in `AddNewSparkClusterAction`. You can check the definition below:
```
            override fun getSparkClusterType(): SparkClusterType {
                return SparkClusterType.SQL_BIG_DATA_CLUSTER
            }

```

---
In reply to: [237105606](https://github.com/Microsoft/azure-tools-for-java/pull/2235#discussion_r237105606) [](ancestors = 237105606)